src/OrleansProviders/Storage/MemoryStorage.cs

```diff
@@ -108,7 +108,7 @@ public virtual Task Init(string name, IProviderRuntime providerRuntime, IProvide
             for (int i = 0; i < numStorageGrains; i++)
             {
                 int idx = i; // Capture variable to avoid modified closure error
-                storageGrains[idx] = new Lazy<IMemoryStorageGrain>(() => MemoryStorageGrainFactory.GetGrain(idx));
+                storageGrains[idx] = new Lazy<IMemoryStorageGrain>(() => providerRuntime.GrainFactory.GetGrain<IMemoryStorageGrain>(idx));
```

Comment:
You don't think that instead we should make GrainFactory be a property of IProvider, so all providers can work just like now, fully backward compat? And still support DI, just like grain does?